File: Dataset/Leetcode/train/55/192.cpp
class Solution {
public:
    bool XXX(vector<int>& nums) {
        int len = nums.size();
        int far = 0;

        for (int i=0; i<len-1; i++) {
            far = max(nums[i]+i, far);
            if (far == i) return false;
        }

        return true;
    }
};

File: Dataset/Leetcode/train/53/192.py
class Solution:
    def XXX(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [0] * n
        dp[0] = nums[0]
        res = nums[0]
        for i in range(1, n):
            dp[i] = max(dp[i-1] + nums[i], nums[i])
            res = max(dp[i], res)
        return res

